queries.sql

```diff
@@ -136,4 +136,84 @@ FROM onwers o
 LEFT JOIN animals a ON o.id = a.owner_id
 GROUP BY o.full_name
 ORDER BY total_owned DESC
+LIMIT 1;
+
+-- Join tables part
+
+SELECT a.name AS last_animal_seen
+FROM visits v
+JOIN animals a ON v.animal_id = a.id
+JOIN vets vt ON v.vet_id = vt.id
+WHERE vt.name = 'Vet William Tatcher'
+ORDER BY v.visit_date DESC
+LIMIT 1;
+
+-- 2- How many different animals did Stephanie Mendez see?
+SELECT COUNT(DISTINCT v.animal_id) AS total_animals_seen
+FROM visits v
+JOIN vets vt ON v.vet_id = vt.id
+WHERE vt.name = 'Vet Stephanie Mendez';
+
+-- 3- List all vets and their specialties, including vets with no specialties.
+SELECT v.name AS vet_name, COALESCE(specialty, 'No Specialty') AS specialty
+FROM vets v
+LEFT JOIN (
+    SELECT vet_id, STRING_AGG(species.name, ', ') AS specialty
+    FROM specializations
+    JOIN species ON specializations.species_id = species.id
+    GROUP BY vet_id
+) s ON v.id = s.vet_id
+ORDER BY vet_name, specialty;
+
+-- 4- List all animals that visited Stephanie Mendez between April 1st and August 30th, 2020.
+SELECT a.name AS animal_name, v.visit_date
+FROM visits v
+JOIN animals a ON v.animal_id = a.id
+WHERE v.vet_id = 3
+  AND v.visit_date BETWEEN '2020-04-01' AND '2020-08-30';
+
+-- 5- What animal has the most visits to vets?
+SELECT a.name AS animal_name, COUNT(*) AS visit_count
+FROM visits v
+JOIN animals a ON v.animal_id = a.id
+GROUP BY a.name
+ORDER BY visit_count DESC
+LIMIT 1;
+
+-- 6- Who was Maisy Smith's first visit?
+SELECT vet.name AS vet_name, MIN(v.visit_date) AS first_visit_date
+FROM visits v
+JOIN vets vet ON v.vet_id = vet.id
+JOIN animals a ON v.animal_id = a.id
+WHERE a.owner_id = (SELECT id FROM onwers WHERE full_name = 'Maisy Smith')
+GROUP BY vet.name
+ORDER BY first_visit_date
+LIMIT 1;
+
```

Comment:
- [ ] Great work on your progress in writing queries based on the project requirements! 👏 However, I'd like to provide some feedback to help you improve the accuracy of your results.

  - Upon reviewing the data you've inserted into the `visits` table, it's evident that `Maisy Smith's` first visit was recorded as `boarmon` when sorted by date. This should be reflected in your query's output. Unfortunately, your current query doesn't return `boarmon` as `Maisy Smith's` first visit.

  - To ensure the correctness of your query, please take another look at it and make the necessary adjustments to obtain an accurate result. Additionally, don't forget to update the `screenshot` of the query result in your pull request summary to reflect the corrected outcome.

Happy coding! 😊🚀🍀

Reply:
> * Upon reviewing the data you've inserted into the `visits` table, it's evident that `Maisy Smith's` first visit was recorded as `boarmon` when sorted by date. This should be reflected in your query's output. Unfortunately, your current query doesn't return `boarmon` as `Maisy Smith's` first visit.

![Uploading image.png…]()